baseplate/metrics.py

```diff
@@ -260,60 +269,63 @@ def stop(self):
         self.send(elapsed)
         self.stopped = True
 
-    def send(self, elapsed):
+    def send(self, elapsed: float) -> None:
         """Directly send a timer value without having to stop/start.
 
         This can be useful when the timing was managed elsewhere and we just
         want to report the result.
 
-        :param elapsed float: The elapsed time in seconds to report.
+        :param elapsed: The elapsed time in seconds to report.
 
         """
-        serialized = self.name + (":{:g}|ms".format(elapsed * 1000.).encode())
+        serialized = self.name + (":{:g}|ms".format(elapsed * 1000.0).encode())
         self.transport.send(serialized)
 
-    def __enter__(self):
+    def __enter__(self) -> None:
         self.start()
 
-    def __exit__(self, exc_type, value, traceback):
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        value: Optional[BaseException],
+        traceback: Optional[TracebackType],
+    ) -> bool:
         self.stop()
+        return False  # don't swallow exception
 
 
 class Counter:
     """A counter for counting events over time."""
 
-    def __init__(self, transport, name):
+    def __init__(self, transport: Transport, name: bytes):
         self.transport = transport
         self.name = name
 
-    def increment(self, delta=1, sample_rate=1.0):
+    def increment(self, delta: float = 1, sample_rate: float = 1.0) -> None:
```

Comment:
if delta is a float, should the default be `1.0`?

Reply:
It looks like this happens in several places.

Reply:
Or should `delta` be `int`?

Reply:
delta should be float, but for typing purposes ints count as floats. i don't know that it matters, but i'll  go be more precise about this.